streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -80,11 +80,13 @@
      *          |      +-----+-------+
      *          +<---- | Starting (1)|----->+
      *          |      +-----+-------+      |
-     *          |            |              |
-     *          |            |              |
-     *          |            v              |
-     *          |      +-----+-------+      |
-     *          +<---- | Partitions  |      |
+     *          |                           |
+     *          |                           |
+     *          |            +<----------+  |
+     *          |            |           |  |
+     *          |            v           |  |
+     *          |      +-----+-------+   |  |
+     *          +<---- | Partitions  | --+  |
```

Comment:
This valid loop was only mentioned in the notes but was not shown in the diagram.

Reply:
Nice catch